libsql/src/sync.rs

```diff
@@ -1,25 +1,39 @@
 use crate::Result;
 
+const METADATA_VERSION: u32 = 0;
+
 const DEFAULT_MAX_RETRIES: usize = 5;
 pub struct SyncContext {
+    db_path: String,
     sync_url: String,
     auth_token: Option<String>,
     max_retries: usize,
+    /// Represents the max_frame_no from the server.
     durable_frame_num: u32,
 }
 
 impl SyncContext {
-    pub fn new(sync_url: String, auth_token: Option<String>) -> Self {
-        Self {
+    pub async fn new(
+        db_path: String,
+        sync_url: String,
+        auth_token: Option<String>,
+    ) -> Result<Self> {
+        let mut me = Self {
+            db_path,
             sync_url,
             auth_token,
             durable_frame_num: 0,
             max_retries: DEFAULT_MAX_RETRIES,
-        }
+        };
+
+        me.read_metadata().await?;
+
+        Ok(me)
     }
 
+    #[tracing::instrument(skip(self, frame))]
```

Comment:
What does this do?

Reply:
This attaches a tracing span to each call. So this adds a span that includes the frame_no and generation being sent and any log that is produced within this fn (and span) will have that data attached allowing us to simplify the actual log lines.